python-avd/pyavd/_cv/workflows/finalize_workspace_on_cv.py

```diff
@@ -87,3 +91,50 @@ async def finalize_workspace_on_cv(workspace: CVWorkspace, cv_client: CVClient)
         return
 
     return
+
+
+async def process_workspace_build_errors(
+    workspace_id: str,
+    build_id: str,
+    cv_client: CVClient,
+) -> list[CVWSBuildError]:
+    """
+    Get and parse Workspace Build Details Response.
+
+    Extract the following errors:
+    - Config validation errors.
+    - TODO: Image validation errors.
+
+    Parameters:
+        workspace_id: Unique identifier the workspace.
+        build_id: Unique identifier of the last WS build attempt.
+        cv_client: CVClient.
+
+    Returns:
+        List of CVWSBuildError objects describing observed WS build validation errors.
+    """
+    try:
+        ws_build_details = await cv_client.get_workspace_build_details(workspace_id=workspace_id, build_id=build_id)
+
+        ws_build_errors = [
+            {
+                "serial_number": response.value.key.device_id,
```

Comment:
Let's think about the use cases. Maybe it will be better to have a list of config errors and a list of other errors. To make it easier to parse.
Seing the model you get from CV, I think I would just return `config_validation_results` as the list. An item per device with sub lists of warnings and errors.
Feel free to push back. Maybe the use of this will make it clear.

Reply:
Tried to implement this model in 75c4de28a